packages/web/src/components/search/DataSearch.js

```diff
@@ -631,15 +631,15 @@ class DataSearch extends Component {
 		checkValueChange(props.componentId, value, props.beforeValueChange, performUpdate);
 	};
 
-	handleTextChange = debounce((value) => {
+	handleTextChange = (value) => {
 		if (this.props.autosuggest) {
 			this.updateDefaultQuery(value, this.props);
 		} else {
 			this.updateQuery(value, this.props);
 		}
-	}, this.props.debounce);
+	};
 
-	updateDefaultQuery = (value, props) => {
+	updateDefaultQuery = debounce((value, props) => {
```

Comment:
@mohdashraf010897 I think should only use `debounce` wrapper when `debounce` prop is defined. The `debounce` method might be using the `setTimeout` which can create side effects when `debounce` is zero. You can also handle it in `debounce` method to return the same method if `debounce` is `zero`|`undefined`|`null`.